### PR TITLE
Transporter actions

### PR DESCRIPTION
**Pull request by Paola-Xaq**

### CODE REVIEW

**Comment by JRojas999** [comment]
You should include checkstyle test.

File: src/main/java/core/selenium/WebElementActions.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public class WebElementActions{
+
+    private WebDriver driver;
+    private WebDriverWait wait;
+
+    public WebElementActions() {
+        driver = DriverManager.getInstance().getDriver();
+        wait = DriverManager.driverManager.getWait();
+    }
+
```
Comment (cristianfchq) [inline src/main/java/core/selenium/WebElementActions.java:17]: I think that is missing javadocs

File: src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    @FindBy(name = "MainCompetitors__c")
+    private WebElement mainCompetitorInput;
+
+    @FindBy(name = "CurrentGenerators__c")
+    private WebElement currentGeneratorsInput;
+
+    @FindBy(name = "TrackingNumber__c")
+    private WebElement trackingNumberInput;
+
+    //Delivery
```
Comment (cristianfchq) [inline src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java]: this line should be removed

File: src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    public void setNextStep(final String nextStepText) {
+        webElementAction.setInputField(nextStepInput, nextStepText);
+    }
+
+    public void setStage() {
+        webElementAction.clickDropdown(stageInput, optionStageDropdown);
+    }
+    //
```
Comment (cristianfchq) [inline src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java]: this line should be removed

File: src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    public void setProbability(final String probabilityText) {
+        webElementAction.setInputField(probabilityInput, probabilityText);
+    }
+
+    //
```
Comment (cristianfchq) [inline src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java]: this line should be removed

File: src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    public void setCurrentGenerator(final String currentGeneratorText) {
+        webElementAction.setInputField(currentGeneratorsInput, currentGeneratorText);
+    }
+
+    public void setTrackingNumer(final String trackingNumerText) {
+        webElementAction.setInputField(trackingNumberInput, trackingNumerText);
+    }
+
+    //
```
Comment (cristianfchq) [inline src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java]: this line should be removed

**Review by cristianfchq — APPROVED** [review]

File: src/main/java/core/selenium/DriverManager.java [hunk 6]
```diff
@@ -1,30 +1,62 @@
...
+    public WebDriverWait getWait() {
+        return wait;
+    }
+
+    private void setDriverManagerType() {
+        String browser = ConfigEnvVar.getInstance().getBrowser();
+        DriverManagerType driverManagerType = DriverManagerType.valueOf(browser);
+        WebDriverManager.getInstance(driverManagerType).setup();
+        switch (driverConfig.getBrowser()) {
+            case "Firefox":
```
Comment (jpgonzales27) [inline src/main/java/core/selenium/DriverManager.java:50]: Pao pao, i think this method shoud be change for a factorey

File: src/main/java/core/selenium/drivers/ChromeDriverOption.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+public class ChromeDriverOption {
+
+    /**
+     * Sets the options for the Chrome Webdriver option.
+     * @return Webdriver option.
+     */
+    public WebDriver setChromeDriver() {
+        ChromeOptions options = new ChromeOptions();
+        Map<String, Object> preferences = new HashMap<>();
+        preferences.put("profile.default_content_setting_values.notifications", 2);
```
Comment (jpgonzales27) [inline src/main/java/core/selenium/drivers/ChromeDriverOption.java:20]: Maybe this string you should move into a constant

File: src/main/java/salesforce/ui/pages/LoginPage.java [hunk 8]
```diff
@@ -9,20 +8,12 @@
     @FindBy(name = "username")
     private WebElement usernameTxtBox;
 
-    @FindBy(name = "password")
+    @FindBy(name = "pw")
```
Comment (jpgonzales27) [inline src/main/java/salesforce/ui/pages/LoginPage.java:11]: PAO is this locator right? I think the locator name is password

File: src/main/java/salesforce/ui/pages/opportunity/CreatedOpportunity.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package salesforce.ui.pages.opportunity;
+
+import salesforce.ui.pages.BasePage;
+
+public class CreatedOpportunity extends BasePage {
+    @Override
+    protected void waitForPageLoaded() {
+
```
Comment (jpgonzales27) [inline src/main/java/salesforce/ui/pages/opportunity/CreatedOpportunity.java:8]: this method should not be empty

File: src/test/java/salesforce/opoortunity/OpportunityTests.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+public class OpportunityTests extends BaseTest {
+
+    private OpportunityPage opportunityPage;
+    private NewOpportunityPage formOpportunity;
+    private HomePage homePage;
+    private PageTransporter pageTransporter = new PageTransporter();
+
+    @Test
+    public void testCreateOpportunityWithRequiredValuesOnly() {
+        String username = ConfigEnvVar.getInstance().getUserName();
```
Comment (jpgonzales27) [inline src/test/java/salesforce/opoortunity/OpportunityTests.java:21]: Maybe you can put these configurations in the before method

File: src/main/java/core/selenium/DriverManager.java (hunk 6 above)
Comment (Paola-Xaq) [reply src/main/java/core/selenium/DriverManager.java:50]: yeah, I will try to implement it in the next PR, thanks

File: src/main/java/core/selenium/WebElementActions.java (hunk 1 above)
Comment (Paola-Xaq) [reply src/main/java/core/selenium/WebElementActions.java:17]: yeah I already add them in the other PR

File: src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java (hunk 2 above)
Comment (Paola-Xaq) [reply src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java]: done!

File: src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java (hunk 3 above)
Comment (Paola-Xaq) [reply src/main/java/salesforce/ui/pages/opportunity/NewOpportunityPage.java]: done as well

File: src/main/java/salesforce/ui/pages/LoginPage.java (hunk 8 above)
Comment (Paola-Xaq) [reply src/main/java/salesforce/ui/pages/LoginPage.java:11]: yeah.. I don't know why .. it didn't work with password

File: src/main/java/salesforce/ui/pages/opportunity/CreatedOpportunity.java (hunk 9 above)
Comment (Paola-Xaq) [reply src/main/java/salesforce/ui/pages/opportunity/CreatedOpportunity.java:8]: its fixed in the next PR

File: src/test/java/salesforce/opoortunity/OpportunityTests.java (hunk 10 above)
Comment (Paola-Xaq) [reply src/test/java/salesforce/opoortunity/OpportunityTests.java:21]: its already fix

**Pull request merged**